app/controllers/organisations/detail.js

```diff
@@ -1,6 +1,94 @@
 import Ember from "ember";
 
-export default Ember.Controller.extend({
-  gcOrganisationUsers: null,
-  backLinkPath: ""
+import SearchOptionMixin from "stock/mixins/search_option";
+import AsyncMixin, { ERROR_STRATEGIES } from "stock/mixins/async";
+
+export default Ember.Controller.extend(SearchOptionMixin, AsyncMixin, {
+  organisationService: Ember.inject.service(),
+  organisation: Ember.computed.alias("model"),
+
+  name_en_error: Ember.computed("model.nameEn", "validate", function() {
+    return this.get("validate") && !this.get("model.nameEn").trim().length;
+  }),
+
+  country_error: Ember.computed("countryValue", "validate", function() {
+    return this.get("validate") && !this.get("countryValue");
+  }),
+
+  website_error: Ember.computed("website", "validate", function() {
+    const websiteRegEx = new RegExp(
+      `^(www\.|https?:\/\/(www\.)?)[a-zA-Z0-9-]+\.[a-zA-Z]+\.?[a-zA-Z0-9-#.]*[a-z]$`
+    );
+
+    return this.get("website") && !this.get("website").match(websiteRegEx);
+  }),
+
+  actions: {
+    /**
+     * Updates the organisation if
+     *      nameEn is present
+     *      country is present
+     *      type is present
+     *      website has a valid format iff its present
+     */
+    updateOrganisation() {
+      const changedAttrs = this.get("model").changedAttributes();
+      Object.keys(changedAttrs).map(attr => {
+        switch (attr) {
+          case "nameEn":
+            this.set("validate", true);
+        }
+      });
+
+      if (
+        !this.get("name_en_error") &&
+        !this.get("country_error") &&
+        !this.get("website_error")
+      ) {
+        this.send("update", this.get("model"));
+      }
```

Comment:
Shouldn't you turn on "validate" every time ? 🤔

Reply:
Btw, I recommend not having the error properties depend on the `"validate"` flag. They should always return the error if present. 

The reason for that is that some other developer might come in one day and call `this.get("name_en_error")` with the expectation that he/she will get the true error state. Which could lead to some drama : )

Maybe you can turn on a `"showErrors`" property whenever you want the UI to show the errors

Reply:
done : )